sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java

```diff
@@ -0,0 +1,155 @@
+import org.apache.spark.sql.catalyst.analysis.NoSuchViewException;
+import org.apache.spark.sql.catalyst.analysis.ViewAlreadyExistsException;
+import org.apache.spark.sql.types.StructType;
+
+/**
+ * Catalog methods for working with views.
+ */
+@DeveloperApi
+public interface ViewCatalog extends CatalogPlugin {
+
+  /**
+   * List the views in a namespace from the catalog.
+   * <p>
+   * If the catalog supports tables, this must return identifiers for only views and not tables.
+   *
+   * @param namespace a multi-part namespace
+   * @return an array of Identifiers for views
+   * @throws NoSuchNamespaceException If the namespace does not exist (optional).
+   */
+  Identifier[] listViews(String... namespace) throws NoSuchNamespaceException;
+
+  /**
+   * Load view metadata by {@link Identifier ident} from the catalog.
+   * <p>
+   * If the catalog supports tables and contains a table for the identifier and not a view,
+   * this must throw {@link NoSuchViewException}.
+   *
+   * @param ident a view identifier
+   * @return the view description
+   * @throws NoSuchViewException If the view doesn't exist or is a table
+   */
+  View loadView(Identifier ident) throws NoSuchViewException;
+
+  /**
+   * Invalidate cached view metadata for an {@link Identifier identifier}.
+   * <p>
+   * If the view is already loaded or cached, drop cached data. If the view does not exist or is
+   * not cached, do nothing. Calling this method should not query remote services.
+   *
+   * @param ident a view identifier
+   */
+  default void invalidateView(Identifier ident) {
+  }
+
+  /**
+   * Test whether a view exists using an {@link Identifier identifier} from the catalog.
+   * <p>
+   * If the catalog supports views and contains a view for the identifier and not a table,
+   * this must return false.
+   *
+   * @param ident a view identifier
+   * @return true if the view exists, false otherwise
+   */
+  default boolean viewExists(Identifier ident) {
+    try {
+      return loadView(ident) != null;
+    } catch (NoSuchViewException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Create a view in the catalog.
+   *
+   * @param ident a view identifier
+   * @param sql the SQL text that defines the view
+   * @param currentCatalog the current catalog
+   * @param currentNamespace the current namespace
+   * @param schema the view query output schema
+   * @param columnAliases the column aliases
+   * @param columnComments the column comments
+   * @param properties the view properties
+   * @return the view created
+   * @throws ViewAlreadyExistsException If a view or table already exists for the identifier
+   * @throws NoSuchNamespaceException If the identifier namespace does not exist (optional)
+   */
+  View createView(
+      Identifier ident,
+      String sql,
+      String currentCatalog,
+      String[] currentNamespace,
+      StructType schema,
+      String[] columnAliases,
```

Comment:
We need `queryColumnNames` here as well right?

Reply:
Should we create a Builder?

Reply:
Actually, should the argument to this just be a `View`...? We are basically passing all of the same information here. `View` doesn't contain any "derived" fields you can only learn after creating the view. Similar to `ViewChange`.

Reply:
I'd hesitate to do so because these have different meanings: one is more like "CreateViewRequest", the other like "ViewMetadata". Using the same interface may cause confusion and evolving one of them in the future.

Reply:
How about builder?
```
public interface ViewBuilder {

  ViewBuilder withQuery(String query);
  ViewBuilder withCurrentCatalog(String defaultCatalog);
  ViewBuilder withCurrentNamespace(String[] defaultNamespaces);
  ViewBuilder withSchema(StructType schema);
  ViewBuilder withQueryColumnNames(String[] queryColumnNames);
  ViewBuilder withColumnAliases(String[] columnAliases);
  ViewBuilder withColumnComments(String[] columnComments);
  ViewBuilder withProperties(Map<String, String> properties);
  ViewBuilder withProperty(String key, String value);
  View create();
  View replace();
  View createOrReplace();
}

ViewCatalog {
  ViewBuilder buildView(Identifier ident);
}
```